src/components/forms/Datepicker.tsx

```diff
@@ -8,22 +8,277 @@
  * @license https://www.elastic.co/licensing/elastic-license
  */
 
-import { useState } from 'react';
-import ReactDatePicker from 'react-datepicker';
+import {
+  createElement,
+  forwardRef,
+  HTMLProps,
+  Ref,
+  useEffect,
+  useState,
+} from 'react';
+import ReactDatePicker, { ReactDatePickerProps } from 'react-datepicker';
 import CommonProps from '../../common/interfaces/common-props.interface';
 import 'react-datepicker/dist/react-datepicker.css';
+import { useTranslation } from 'react-i18next';
+import { Icon } from '../icons/Icon';
+import {
+  MdCalendarMonth,
+  MdKeyboardArrowLeft,
+  MdKeyboardArrowRight,
+} from 'react-icons/md';
+import { useAccentColor } from '$app/common/hooks/useAccentColor';
+import dayjs from 'dayjs';
+import { useCurrentCompanyDateFormats } from '$app/common/hooks/useCurrentCompanyDateFormats';
+import { date as formatDate } from '$app/common/helpers';
+import { Button } from './Button';
+import { SelectField } from './SelectField';
+import { useAtomValue } from 'jotai';
+import { dayJSLocaleAtom } from '$app/App';
 
-type Props = CommonProps;
+interface Props extends CommonProps {
+  required?: boolean;
+  onValueChange?: (value: string) => unknown;
+  minDate?: string;
+}
+
+interface Month {
+  value: number;
+  label: string;
+}
+
+const MONTHS: Month[] = [
+  {
+    value: 0,
+    label: 'january',
+  },
```

Comment:
Are we sure we want to index months by 0?

Does this affect the output (on change handler)?

Reply:
@beganovich No worries, I tested it and after your comment I tested it again, it works as expected. Actually, I was also surprised to see that months in Javascript start from 0, but not from 1. So it's just adapted to the JS logic.